File: cpp/54_字符流中第一个不重复的字符.cpp
class Solution
{
public:
    string s; // 用来记录顺序
    char hash[256] = {0};
    
    // Insert one char from stringstream
    void Insert(char ch) {
        s += ch;
        hash[ch]++;
    }
    
    // return the first appearence once char in current stringstream
    char FirstAppearingOnce() {
        for (auto i : s) {
            if (hash[i] == 1) {
                return i;
            }
        }
        
        return '#';
    }
};

File: python/54_字符流中第一个不重复的字符.py
#!/bin/bash/ python3
#-*- coding:utf-8 -*-

class Solution:
    # 返回对应char
    def __init__(self):
        self.hash = []
    
    def Insert(self, ch):
        if ch != None:
            self.hash.append(ch)
    
    def FirstAppearingOnce(self):
        for i in self.hash:
            if self.hash.count(i) == 1:
                return i
        return "#"